services/management/mongodb.go

```diff
@@ -72,6 +72,20 @@ func (s *MongoDBService) Add(ctx context.Context, req *managementpb.AddMongoDBRe
 
 		mongoDBOptions := models.MongoDBOptionsFromRequest(req)
 
+		switch {
+		case req.AddNode != nil && req.AddNode.NodeType == inventorypb.NodeType_REMOTE_NODE:
+			req.MetricsMode = managementpb.MetricsMode_PULL
```

Comment:
The same problem everywhere: we should not overwrite mode before checking that it is set to auto. If user requested push or pull explicitly, and this mode is not supported in this particular case, we should return error.

Reply:
/cc @BupycHuk

Reply:
Agree